### PR TITLE
Add functionality to retrieve all pending items from pools 

### PR DESCRIPTION
**Pull request by 0xKiwi**
**What type of PR is this?**
This PR adds functionality to the slashings and exits pools to allow retrieving all pending items, as their only available action prior was to return only the max allowed per block.

This PR is needed for #7525

### CODE REVIEW

File: beacon-chain/operations/slashings/service.go [hunk 1]
```diff
@@ -26,8 +25,9 @@ func NewPool() *Pool {
 }
 
 // PendingAttesterSlashings returns attester slashings that are able to be included into a block.
-// This method will not return more than the block enforced MaxAttesterSlashings.
-func (p *Pool) PendingAttesterSlashings(ctx context.Context, state *beaconstate.BeaconState) []*ethpb.AttesterSlashing {
+// This method will return all pending attester slashings unless the `block` parameter is set to true
+// to indicate the request is for a block.
+func (p *Pool) PendingAttesterSlashings(ctx context.Context, state *beaconstate.BeaconState, block bool) []*ethpb.AttesterSlashing {
```
Comment (rauljordan) [inline beacon-chain/operations/slashings/service.go:30]: block bool is ambiguous, it sounds like you want the function to block. Perhaps we can call for `forProposal`? or `forBlockProposal`?

Comment (rauljordan) [reply beacon-chain/operations/slashings/service.go:30]: Maybe we can call it `noLimit` ?

File: beacon-chain/operations/slashings/service_proposer_test.go [hunk 2]
```diff
@@ -407,7 +416,7 @@ func TestPool_PendingProposerSlashings_Slashed(t *testing.T) {
 			p := &Pool{
 				pendingProposerSlashing: tt.fields.pending,
 			}
-			assert.DeepEqual(t, tt.want, p.PendingProposerSlashings(context.Background(), beaconState))
+			assert.DeepEqual(t, tt.want, p.PendingProposerSlashings(context.Background(), beaconState, false /*block*/))
```
Comment (rauljordan) [inline beacon-chain/operations/slashings/service_proposer_test.go:419]: Change comment here if we change the input arg name to something else

File: beacon-chain/operations/slashings/service.go [hunk 3]
```diff
@@ -37,12 +37,16 @@ func (p *Pool) PendingAttesterSlashings(ctx context.Context, state *beaconstate.
...
-	pending := make([]*ethpb.AttesterSlashing, 0, mathutil.Min(uint64(len(p.pendingAttesterSlashing)), params.BeaconConfig().MaxAttesterSlashings))
+
+	// Allocate pending slice with a capacity of maxAttesterSlashings or len(p.pendingAttesterSlashing)) depending on the request.
+	maxSlashings := params.BeaconConfig().MaxAttesterSlashings
+	if !block {
+		maxSlashings = uint64(len(p.pendingAttesterSlashing))
+	}
+	pending := make([]*ethpb.AttesterSlashing, 0, maxSlashings)
 	for i := 0; i < len(p.pendingAttesterSlashing); i++ {
 		slashing := p.pendingAttesterSlashing[i]
```
Comment (terencechain) [inline beacon-chain/operations/slashings/service.go:48]: Can you move `slashing := ...` after the break check?

File: beacon-chain/operations/slashings/service.go [hunk 4]
```diff
@@ -78,12 +83,15 @@ func (p *Pool) PendingProposerSlashings(ctx context.Context, state *beaconstate.
...
-	// since the array cannot exceed the max and is typically less than the max value.
-	pending := make([]*ethpb.ProposerSlashing, 0, mathutil.Min(uint64(len(p.pendingProposerSlashing)), params.BeaconConfig().MaxProposerSlashings))
+	// Allocate pending slice with a capacity of len(p.pendingProposerSlashing) or maxProposerSlashings depending on the request.
+	maxSlashings := params.BeaconConfig().MaxProposerSlashings
+	if !block {
+		maxSlashings = uint64(len(p.pendingProposerSlashing))
+	}
+	pending := make([]*ethpb.ProposerSlashing, 0, maxSlashings)
 	for i := 0; i < len(p.pendingProposerSlashing); i++ {
 		slashing := p.pendingProposerSlashing[i]
```
Comment (terencechain) [inline beacon-chain/operations/slashings/service.go:93]: Can you move `slashing := ...` after the break check?

File: beacon-chain/operations/slashings/service.go [hunk 5]
```diff
@@ -26,25 +25,30 @@ func NewPool() *Pool {
 }
 
 // PendingAttesterSlashings returns attester slashings that are able to be included into a block.
-// This method will not return more than the block enforced MaxAttesterSlashings.
-func (p *Pool) PendingAttesterSlashings(ctx context.Context, state *beaconstate.BeaconState) []*ethpb.AttesterSlashing {
-	p.lock.RLock()
```
Comment (0xKiwi) [inline beacon-chain/operations/slashings/service.go:31]: Not sure why this was a RLock when this functin mutates.

**Review by rauljordan — APPROVED** [review]

**Pull request merged**